hack/generated/pkg/reconcilers/azure_deployment_reconciler.go

```diff
@@ -108,20 +108,56 @@ func NewAzureDeploymentReconciler(
 	}
 }
 
+// checkForFatalReconciliationError:
+// - at the moment this is only raised during tests if we determine that a record-replay
+//   test has gone off the rails, but it will be useful for scenarios in the real world as well
+func (r *AzureDeploymentReconciler) checkForFatalReconciliationError(ctx context.Context, err error) error {
+	var fatal FatalReconciliationError
+	if !errors.As(err, &fatal) {
+		return err
+	}
+
+	condition := r.PositiveConditions.MakeFalseCondition(
```

Comment:
This is weird? Is transiting through `PositiveConditions` the only way to make a negative one?

Reply:
quote from `positive_condition_builder`:

> PositiveConditionBuilder makes positive polarity conditions. A positive polarity condition is a condition where Status==True is the "happy path" state, and Status==False is a failure state.

Our condition "Ready" is positive; this sets Ready to False.